sdk/eventhub/Azure.Messaging.EventHubs/assets/live-tests-azure-setup.ps1

```diff
@@ -200,113 +73,43 @@ if ($Help)
   exit 0
 }
 
-if ([String]::IsNullOrEmpty($AzureRegion))
-{
-    $AzureRegion = "southcentralus"
-}
-
-# Disallow principal names with a space.
-
-if ($ServicePrincipalName.Contains(" "))
-{
-    Write-Error "The principal name may not contain spaces."
-    exit -1
-}
-
-# Verify the location is valid for an Event Hubs namespace.
-
-$validLocations = @{}
-Get-AzLocation | where { $_.Providers.Contains("Microsoft.EventHub")} | ForEach { $validLocations[$_.Location] = $_.Location }
-
-if (!$validLocations.Contains($AzureRegion))
-{
-    Write-Error "The Azure region must be one of: `n$($validLocations.Keys -join ", ")`n`n" 
-    exit -1
-}
-
-# Capture the subscription.  The cmdlet will error if there was no subscription, 
-# so no need to validate.
-
-Write-Host ""
-Write-Host "Working:"
-Write-Host "`t...Requesting subscription"
-$subscription = (Get-AzSubscription -SubscriptionName "$($SubscriptionName)" -ErrorAction SilentlyContinue)
-
-if ($subscription -eq $null)
-{
-    Write-Error "Unable to locate the requested Azure subscription: $($SubscriptionName)"
-    exit -1
-}
-
-Set-AzContext -SubscriptionId "$($subscription.Id)" -Scope "Process" | Out-Null
-
-# Create the resource group, if needed.
-
-Write-Host "`t...Requesting resource group"
-
-$createResourceGroup = $false
-$resourceGroup = (Get-AzResourceGroup -ResourceGroupName "$($ResourceGroupName)" -ErrorAction SilentlyContinue)
-
-if ($resourceGroup -eq $null)
+if ([string]::IsNullOrEmpty($azureRegion)) 
 {
-    $createResourceGroup = $true
+  $azureRegion = "southcentralus"
 }
 
-if ($createResourceGroup)
-{
-    Write-Host "`t...Creating new resource group"
-    $resourceGroup = (New-AzResourceGroup -Name "$($ResourceGroupName)" -Location "$($AzureRegion)")
-}
-
-if ($resourceGroup -eq $null)
-{
-    Write-Error "Unable to locate or create the resource group: $($ResourceGroupName)"
-    exit -1
-}
+ValidateParameters -ServicePrincipalName "$($servicePrincipalName)" -AzureRegion "$($azureRegion)"
+$subscription = GetSubscriptionAndSetAzureContext -SubscriptionName "$($subscriptionName)"
+CreateResourceGroupIfMissing -ResourceGroupName "$($resourceGroupName)" -AzureRegion "$($azureRegion)"

```

Comment:
This should probably return a flag indicating whether or not it created the resource group.    That way, we can use that as a signal for whether or not we need to clean it up on failure.